src/vs/server/main.js

```diff
@@ -129,20 +129,84 @@ function start() {
 }
 
 /**
+ * If `--pick-port` and `--port` is specified, connect to that port.
+ *
+ * If not and a port range is specified through `--pick-port`
+ * then find a free port in that range. Throw error if no
+ * free port available in range.
+ *
+ * If only `--port` is provided then connect to that port.
+ *
+ * In absence of specified ports, connect to port 8000.
  * @param {string | undefined} strPort
- * @returns {number}
+ * @param {string | undefined} strPickPort
+ * @returns {Promise<number>}
+ * @throws
  */
-function parsePort(strPort) {
-	try {
-		if (strPort) {
-			return parseInt(strPort);
+async function parsePort(strPort, strPickPort) {
+	let specificPort = -1;
+
+	if (strPort) {
+		const port = parseInt(strPort, 10);
+		if (!isNaN(port)) {
+			specificPort = port;
+		} else {
+			console.log('Port is not a number, will default to 8000 if no pick-port is given.');
 		}
-	} catch (e) {
-		console.log('Port is not a number, using 8000 instead.');
 	}
+
+	if (strPickPort) {
+		if (strPickPort.match(/[\d]+-[\d]+/)) {
```

Comment:
Just nitpicking now I guess but you probably want to use `^` and `$` at the beginning and end so that `a1-1a` doesn't match, and the `[]` are not necessary